src/pages/workspaces/workspace/Data.js

```diff
@@ -419,38 +418,175 @@ class WorkspaceDataContent extends Component {
     ])
   }
 
-  renderGlobalVariables() {
-    const { namespace } = this.props
-    const { workspaceAttributes } = this.state
-    const filteredAttributes = _.flow(
-      _.toPairs,
-      _.remove(([key]) => key === 'description' || key.includes(':') || key.startsWith('referenceData-')),
-      _.sortBy(_.first)
-    )(workspaceAttributes)
+  renderLocalVariables() {
+    const { namespace, name, workspace: { accessLevel }, ajax: { Workspaces } } = this.props
+    const { workspaceAttributes, editIndex, deleteIndex, editKey, editValue, editType, addVariableHover, creatingNewVariable } = this.state
+    const canEdit = Utils.canWrite(accessLevel)
+    const stopEditing = () => this.setState(
+      { editIndex: undefined, editKey: undefined, editValue: undefined, editType: undefined, creatingNewVariable: false })
+    const filteredAttributes = [
+      ..._.flow(
+        _.toPairs,
+        _.remove(([key]) => key === 'description' || key.includes(':') || key.startsWith('referenceData-')),
+        _.sortBy(_.first)
+      )(workspaceAttributes), ...(creatingNewVariable ? [['', '']] : [])
+    ]
+    const inputErrors = editIndex && [
```

Comment:
Not using `validate.js` because there are so many variants and they, and their messages, depend on the selected type.